### PR TITLE
kernel: bump 5.15 to 5.15.40

### PR DESCRIPTION
**Pull request by graysky2**
All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <therealgraysky@proton.me>

### CODE REVIEW

**Comment by Boos4721** [comment]
**WTF 5.150**

File: target/linux/generic/pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch [hunk 1]
```diff
@@ -25,9 +25,9 @@ Signed-off-by: Marek Behún <kabel@kernel.org>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1480,6 +1480,18 @@ static void advk_pcie_handle_int(struct
```
Comment (hauke) [inline target/linux/generic/pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch:28]: Please remove this patch it is also upstream, see https://github.com/gregkh/linux/commit/cf7ef0e3029e43f85b525dd49420d56792a4e61f

File: target/linux/generic/backport-5.15/850-v5.17-0007-PCI-aardvark-Mask-all-interrupts-when-unbinding-driv.patch [hunk 2]
```diff
@@ -18,9 +18,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
...
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1693,6 +1693,27 @@ static int advk_pcie_remove(struct platf
- 	val &= ~(PCI_COMMAND_IO | PCI_COMMAND_MEMORY | PCI_COMMAND_MASTER);
- 	advk_writel(pcie, val, PCIE_CORE_CMD_STATUS_REG);
+@@ -1878,6 +1878,27 @@ static int advk_pcie_remove(struct platf
+ 	/* Remove IRQ handler */
+ 	irq_set_chained_handler_and_data(pcie->irq, NULL, NULL);
  
 +	/* Disable MSI */
```
Comment (hauke) [inline target/linux/generic/backport-5.15/850-v5.17-0007-PCI-aardvark-Mask-all-interrupts-when-unbinding-driv.patch:25]: Please remove this patch, it was applied upstream, see https://github.com/gregkh/linux/commit/55e38c42051dc27224c5d27a5d81a8b85c324d0a

File: target/linux/generic/backport-5.15/850-v5.17-0011-PCI-aardvark-Disable-common-PHY-when-unbinding-drive.patch [hunk 3]
```diff
@@ -18,9 +18,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1734,6 +1734,9 @@ static int advk_pcie_remove(struct platf
- 	for (i = 0; i < OB_WIN_COUNT; i++)
- 		advk_pcie_disable_ob_win(pcie, i);
+@@ -1640,6 +1640,9 @@ static int advk_pcie_enable_phy(struct a
```
Comment (hauke) [inline target/linux/generic/backport-5.15/850-v5.17-0011-PCI-aardvark-Disable-common-PHY-when-unbinding-drive.patch:21]: Please remove this patch, it was applied upstream, see https://github.com/gregkh/linux/commit/ac76c36aa417014b4e46d4d1dee7b15c39ccbf8c

File: target/linux/generic/backport-5.15/850-v5.17-0009-PCI-aardvark-Assert-PERST-when-unbinding-driver.patch [hunk 4]
```diff
@@ -20,7 +20,7 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1721,6 +1721,10 @@ static int advk_pcie_remove(struct platf
+@@ -1919,6 +1919,10 @@ static int advk_pcie_remove(struct platf
```
Comment (hauke) [inline target/linux/generic/backport-5.15/850-v5.17-0009-PCI-aardvark-Assert-PERST-when-unbinding-driver.patch:23]: Please remove this patch, it was already applied upstream, see https://github.com/gregkh/linux/commit/246a36b6637740f1be9c3cdd91f022b24b7402e7

File: target/linux/generic/backport-5.15/850-v5.17-0010-PCI-aardvark-Disable-link-training-when-unbinding-dr.patch [hunk 5]
```diff
@@ -20,7 +20,7 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1725,6 +1725,11 @@ static int advk_pcie_remove(struct platf
```
Comment (hauke) [inline target/linux/generic/backport-5.15/850-v5.17-0010-PCI-aardvark-Disable-link-training-when-unbinding-dr.patch:23]: Please remove this patch, it was already applied upstream, see https://github.com/gregkh/linux/commit/a346d08ca45e8475975187f5bab8540cb9ac75d2

File: target/linux/generic/backport-5.15/850-v5.17-0008-PCI-aardvark-Fix-memory-leak-in-driver-unbind.patch [hunk 6]
```diff
@@ -21,9 +21,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/controller/pci-aardvark.c
 +++ b/drivers/pci/controller/pci-aardvark.c
-@@ -1718,6 +1718,9 @@ static int advk_pcie_remove(struct platf
```
Comment (hauke) [inline target/linux/generic/backport-5.15/850-v5.17-0008-PCI-aardvark-Fix-memory-leak-in-driver-unbind.patch:24]: Please remove this patch, it was already applied upstream, see https://github.com/gregkh/linux/commit/af27f54e62243a6609c8e21c98b4e7eed4cde992

**Comment by graysky2** [comment]
@hauke - Thanks for the review.  I made the changes you called out.

File: target/linux/generic/backport-5.15/850-v5.17-0002-PCI-pci-bridge-emul-Add-definitions-for-missing-capa.patch [hunk 7]
```diff
@@ -21,9 +21,9 @@ Signed-off-by: Lorenzo Pieralisi <lorenzo.pieralisi@arm.com>
 
 --- a/drivers/pci/pci-bridge-emul.c
 +++ b/drivers/pci/pci-bridge-emul.c
-@@ -270,6 +270,49 @@ struct pci_bridge_reg_behavior pcie_cap_
- 		.ro = GENMASK(15, 0) | PCI_EXP_RTSTA_PENDING,
- 		.w1c = PCI_EXP_RTSTA_PME,
+@@ -313,6 +313,49 @@ struct pci_bridge_reg_behavior pcie_cap_
```
Comment (hauke) [inline target/linux/generic/backport-5.15/850-v5.17-0002-PCI-pci-bridge-emul-Add-definitions-for-missing-capa.patch:24]: Please remove this patch, it was applied upstream, see https://github.com/gregkh/linux/commit/739c052b9b26bc5abff3f6a5139292c00bc1964e

**Comment by graysky2** [comment]
Done.  I am surprised that the update script missed these.

File: target/linux/generic/pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch (hunk 1 above)
Comment (hauke) [reply target/linux/generic/pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch:28]: This is not resolved

Comment (graysky2) [reply target/linux/generic/pending-5.15/850-0016-PCI-aardvark-Add-support-for-PME-interrupts.patch:28]: Should be now

**Comment by graysky2** [comment]
I only see two patches now and cannot find either of them in the [5.15.39 stable tree](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/?h=v5.15.39):
```
% find . -name '*aardv*.patch'|grep 5.15
./target/linux/generic/pending-5.15/851-0006-Revert-PCI-aardvark-Fix-initialization-with-old-Marv.patch
./target/linux/generic/pending-5.15/850-0023-PCI-aardvark-Make-main-irq_chip-structure-a-static-d.patch
```

**Comment by hauke** [comment]
I did a diff between the new and the old content of the drivers/pci/controller/pci-aardvark.c file. I checked why it was different and found the problems.

Normally when I review these pull requests I closely look at the patches which change other that just changing some offsets.

**Comment by 1Jeff1** [comment]
> I only see two patches now and cannot find either of them in the [5.15.39 stable tree](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/?h=v5.15.39):


@graysky2  850-0023-PCI-aardvark-Make-main-irq_chip-structure-a-static-d.patch https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.39&id=d951cbef2e4607868cc94b86ae6def03ec2fff4a

**Comment by hauke** [comment]
850-0023-PCI-aardvark-Make-main-irq_chip-structure-a-static-d.patch is not included in kernel 5.15.39. 

The commit https://github.com/gregkh/linux/commit/d951cbef2e4607868cc94b86ae6def03ec2fff4a matches this patch: 
target/linux/generic/pending-5.15/850-0006-PCI-aardvark-Make-MSI-irq_chip-structures-static-dri.patch

**Comment by hauke** [comment]
Thank you for your patches, I applied them to master in 6cdf974a3d0cfebf4a37aaf2876b2ece0f11363d and e156f1f862ff0c834df3b92a180598235e578e84.

**Pull request closed without merging**